src/compute-client/src/plan/interpret/api.rs

```diff
@@ -0,0 +1,650 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Utilities for abstract interpretation of [crate::plan::Plan] structures.
+//!
+//! Those can be used to define analysis passes over [crate::plan::Plan]s in a
+//! consistent and unified manner. The primary abstraction here is the
+//! [Interpreter] trait.
+
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+
+use differential_dataflow::lattice::Lattice;
+use itertools::zip_eq;
+use mz_expr::{EvalError, Id, LocalId, MapFilterProject, MirScalarExpr, TableFunc};
+use mz_ore::cast::CastFrom;
+use mz_ore::soft_panic_or_log;
+use mz_repr::{Diff, Row};
+
+use crate::plan::join::JoinPlan;
+use crate::plan::reduce::{KeyValPlan, ReducePlan};
+use crate::plan::threshold::ThresholdPlan;
+use crate::plan::top_k::TopKPlan;
+use crate::plan::{AvailableCollections, GetPlan, Plan};
+
+/// An [abstract interpreter] for [Plan] expressions.
+///
+/// This is an [object algebra] / [tagless final encoding] of the language
+/// defined by [crate::plan::Plan], with the exception of the `Get` and `Let*`
```

Comment:
`Get` is included, so the exception is only `Let*`?

Reply:
Yes, I forgot to update this comment after I figured out that `Get` needs to be part of the interface.